### DISCUSSION TITLE
можно ли присваивать файлам для windows и linux уникальные префиксы?

### DISCUSSION THREAD

**Discussion by kotsmotritnastul** [bol-van/zapret#938] [Q&A] [answered]
запрет для опенврт можно скачать командой `wget $(curl -s https://api.github.com/repos/bol-van/zapret/releases/latest  | jq -r '.assets[] | select(.name | contains ("openwrt")) | .browser_download_url')` чтобы не лазить руками на сайт. 

запрет для винды можно скачать командой `wget $(curl -s https://api.github.com/repos/bol-van/zapret/releases/latest  | jq -r '.assets[] | select(.name | contains ("zip")) | .browser_download_url')`

но вот конкретно для десктопного линукса так не получится - там в имени файла нет никаких уникальных слов для конкретно этого файла под конкретно десктопный линукс.

можно ли в имена файлов релизов для разных платформ засовывать уникальные для каждой конкретной платформы слова которые можно засунуть в команду `wget $(curl -s https://api.github.com/repos/bol-van/zapret/releases/latest  | jq -r '.assets[] | select(.name | contains ("УНИКАЛЬНОЕ_ДЛЯ_ПЛАТФОРМЫ_СЛОВО")) | .browser_download_url')`?
например для релиза на винды можно в имя зип-архива написать `windows`, для tar.gz для десктопного линукса `linux` или ещё что-нибудь уникальное чтобы такая команда однозначно скачивала именно этот файл и никакой больше.

**Comment by bol-van** [comment]
Там совмещенный релиз. все в одном. вариант для openwrt - сокращенный, чтобы поместилось в /tmp на 64 мб RAM 
Другой причины существования embedded варианта нет

**Reply by kotsmotritnastul** [reply]
я понимаю что совмещённый, но возможно можно будет добавить в имя этого архива какое-нибудь слово которое при `jq` запросе и прочих подобных запросах не требующих вручную лазить на сайт гитхаба будет выдёргивать урл именно этого файла и ничто больше?

я ни разу не пользовался github-actions, но может быть там есть такая функция. можно конечно и руками урл копипастить но хотелось бы автоматизировать обновление программы на новую версию по максимуму.

**Reply by bol-van** [reply]
куда ни посмотри, везде лежат релизы с имением версии в названии. 
обычно привязываются к конкретной версии, тк в новой может что-то поменяться, что не сработает в автоматизированном сценарии

**Reply by bol-van** [reply, marked as answer]
`curl -s https://api.github.com/repos/bol-van/zapret/releases/latest  | jq -r '.assets[]  | .browser_download_url' | grep zapret-v[0-9\.]*.tar.gz | head -n 1`

**Reply by kotsmotritnastul** [reply]
да, так действительно сработало, теперь вытаскивается урл неопенврт-линуксового архива. спасибо.
